MapboxVisionTests/RecordCoordinatorTests.swift

```diff
@@ -86,6 +148,71 @@ final class RecordCoordinatorTests: XCTestCase {
             assertionFailure("Directory removing has failed for path: \(path). Error: \(error)")
         }
     }
+
+    private class TestClipRequest {
+        let start: Float
+        let end: Float
+
+        init(start: Float, end: Float) {
+            self.start = start
+            self.end = end
+        }
+    }
+
+    private class RecordDelegate: RecordCoordinatorDelegate {
+        var onRecordingStarted: ((String) -> Void)?
+        var onRecordingStopped: ((RecordingPath) -> Void)?
+
+        func recordingStarted(path: String) {
+            onRecordingStarted?(path)
+        }
+
+        func recordingStopped(recordingPath: RecordingPath) {
+            onRecordingStopped?(recordingPath)
+        }
+    }
+
+    private func performVideoClipping(with requests: [TestClipRequest], _ message: String = "", line: UInt = #line) {
+        let expectation = XCTestExpectation(description: "All the requested clips are successfully cut")
+        let videoSource = MockVideoSource()
+        videoSource.add(observer: coordinator)
+        videoSource.start()
+
+        var didCreateAllClips = false
+
+        let recordDelegate = RecordDelegate()
+        recordDelegate.onRecordingStopped = { path in
+            if let pathUrl = URL(string: path.recordingPath) {
+                let directoryContent = try? FileManager.default.contentsOfDirectory(at: pathUrl, includingPropertiesForKeys: nil, options: [])
+
+                if let directoryContent = directoryContent {
+                    let numberOfVideos = directoryContent.filter {
+                        $0.pathExtension == path.settings.fileExtension
+                    }
+                    .count
+
+                    didCreateAllClips = numberOfVideos == requests.count
+                }
+            }
+            expectation.fulfill()
```

Comment:
Comment for `testVideoClipping` (line 79) say:
As a result we should get the same number of video files on filesystem
Should we then to fulfill the expectation only when `didCreateAllClips == true`?

```suggestion
    if didCreateAllClips {
            expectation.fulfill()
    }
```

WDYT?

Reply:
the expectation doesn't match the test's goal, we can change the way we fulfill the expectation as you suggest but it doesn't make much sense for me

Reply:
after discussion decided that the expectation had incorrect description, fixed